### ISSUE TITLE
False positive: Value 'response' is unsubscriptable

### ISSUE THREAD

**Issue by chrisburr** [PyCQA/pylint#4148]
### Steps to reproduce

Given a file `test.py`:

```python
import ldap3
server = ldap3.Server('aaaa')
connection = ldap3.Connection(server, client_strategy=ldap3.SAFE_SYNC, auto_bind=True)
status, result, response, _ = connection.search("bbbb", "cccc", attributes=["dddd"])
print(len(response))
print(response[0])
```

The code works as expected when replacing the strings valid values for the ldap server (not relevant for linting).

### Current behavior

```
************* Module test
test.py:5:6: E1136: Value 'response' is unsubscriptable (unsubscriptable-object)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.0
astroid 2.5
Python 3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27)
[GCC 9.3.0]
```

Additional dependencies:
```
ldap3                     2.9                pyhd3eb1b0_0
```

**Comment by hippo91** [comment]
@chrisburr thanks for the report. It seems that astroid, the pylint inference engine, needs a brain to handle ldap module.

**Comment by mbyrnepr2** [comment]
I can't reproduce this on Python 3.7, 3.8, 3.9 or 3.10.

There could potentially be other issues we don't know about since I believe astroid does not currently give special treatment to ldap3 in the form of an astroid brain; but that is probably best served in the form of a plugin.

I think we can close but tagging @DanielNoord in case I've missed something!

**Comment by Pierre-Sassoulas** [comment]
Did you install ldap3 @mbyrnepr2 ?

**Comment by mbyrnepr2** [comment]
> ```python
> import ldap3
> server = ldap3.Server('aaaa')
> connection = ldap3.Connection(server, client_strategy=ldap3.SAFE_SYNC, auto_bind=True)
> status, result, response, _ = connection.search("bbbb", "cccc", attributes=["dddd"])
> print(len(response))
> print(response[0])
> ```

Yes indeed I did! Without it I get an import error.
